File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        Stack<Character> stack = new Stack<>();
        for(int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            if(c == '(') stack.push('(');
            else {
                if(!stack.isEmpty()) stack.pop();
                else return false;
            }
        }
        return stack.size() == 0;
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    var cnt = 0;
    for(let i = 0; i < s.length; i++) {
        if(s[i] === '(') {
            cnt++;
        } else {
            cnt--;
            if(cnt < 0) return false;
        }
    }
    return cnt === 0;
}